itests/eth_deploy_test.go

```diff
@@ -140,6 +140,11 @@ func TestDeployment(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, receipt)
 
+	// Then lookup the receipt.
+	receipts, err := client.EthGetBlockReceipts(ctx, ethtypes.EthBlockNumberOrHash{BlockHash: &receipt.BlockHash})
+	require.NoError(t, err)
+	require.NotNil(t, receipts)
+
```

Comment:
```suggestion
	require.Greater(t, len(receipts), 0)
	var matchingReceipt *api.EthTxReceipt
	for _, r := range receipts {
		if r.TransactionHash == receipt.TransactionHash {
			require.Nil(t, matchingReceipt, "Multiple matching receipts found")
			matchingReceipt = r
		}
	}
	require.NotNil(t, matchingReceipt, "No matching receipt found")
	require.Equal(t, receipt, matchingReceipt)

```

The problem is that this fails, and I don't know why exactly. Removing the bytes (which match), this is what we're comparing:

```
&api.EthTxReceipt{TransactionHash:ethtypes.EthHash{}, TransactionIndex:0x0, BlockHash:ethtypes.EthHash{}, BlockNumber:0x8, From:ethtypes.EthAddress{}, To:(*ethtypes.EthAddress)(nil), StateRoot:ethtypes.EthHash{}, Status:0x1, ContractAddress:(*ethtypes.EthAddress)(0xc008adc438), CumulativeGasUsed:0x0, GasUsed:0xf940d3, EffectiveGasPrice:35906732, LogsBloom:ethtypes.EthBytes{}, Logs:[]ethtypes.EthLog{}, Type:0x2}
```

```
&api.EthTxReceipt{TransactionHash:ethtypes.EthHash{}, TransactionIndex:0x0, BlockHash:ethtypes.EthHash{}, BlockNumber:0x8, From:ethtypes.EthAddress{}, To:(*ethtypes.EthAddress)(nil), StateRoot:ethtypes.EthHash{}, Status:0x1, ContractAddress:(*ethtypes.EthAddress)(0xc0088f8630), CumulativeGasUsed:0x0, GasUsed:0xf940d3, EffectiveGasPrice:41000634, LogsBloom:ethtypes.EthBytes{}, Logs:[]ethtypes.EthLog{}, Type:0x2}
```

`ContractAddress:(*ethtypes.EthAddress)` not matching is just a pointer differential and I think we could add a matcher for that one specifically. But `EffectiveGasPrice` isn't matching, it's quite different. They should match, right? Calculated here:

https://github.com/filecoin-project/lotus/blob/5869a173ed8e4419601979316aea32c11d4d8ab7/node/impl/full/eth_utils.go#L745-L754

Any idea why we might be getting different values?

Reply:
```go
	require.NotNil(t, receipt.ContractAddress)
	require.NotNil(t, matchingReceipt.ContractAddress)
	require.Equal(t, *receipt.ContractAddress, *matchingReceipt.ContractAddress)
	originalReceiptContractAddress := receipt.ContractAddress
	receipt.ContractAddress = nil
	matchingReceipt.ContractAddress = nil
	require.Equal(t, receipt, matchingReceipt)
	receipt.ContractAddress = originalReceiptContractAddress
```

that gets around the pointer problem, but the gas mismatch remains